tests/functional/s3-marker.sh

```diff
@@ -47,4 +47,74 @@ ${AWS} s3 rm s3://${BUCKET}/bb
 ${AWS} s3 rm s3://${BUCKET}/dd/dd
 ${AWS} s3 rm s3://${BUCKET}/ff
 
+
+# test marker with prefix on object
+
+for i in $(seq 1 21); do
+    ${AWS} s3 cp /etc/magic s3://${BUCKET}/subdir/object-$i
+done
+
+echo "Recursive listing with default page-size"
+objs=$(${AWS} s3api list-objects --bucket ${BUCKET} | grep -c Key)
+[ $objs -eq 21 ] || exit 1
+
+echo "Recursive listing with page-size 10"
+objs=$(${AWS} s3api list-objects --bucket ${BUCKET} --page-size 10 | grep -c Key)
+[ $objs -eq 21 ] || exit 1
+
+echo "Recursive listing with page-size 10 and prefix with trailing /"
+objs=$(${AWS} s3api list-objects --bucket ${BUCKET} --page-size 10 --prefix subdir/ | grep -c Key)
+[ $objs -eq 21 ] || exit 1
```

Comment:
`--page-size 10` should only gives 10 results, doesn't it ?

Reply:
Nope, it is used for paginate listing internally by `aws`